[native-pos] Disable AsyncDataCache (thus prefetching) (#20969)
prestodb/presto: presto-docs/src/main/sphinx/develop/presto-native.rst

    If the time period is less than or equal to the parameter value, the request
    is valid.
    If the time period exceeds the parameter value, the request is rejected as
    authentication failure (HTTP 401).

Async Data Cache and Prefetching
--------------------------------

> ``num-connector-io-threads``

Reviewer: Is this specific to Hive connector?

^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    * **Type** ``integer``
    * **Default value:** ``30``

    Size of IO executor for connectors to do preload/prefetch.  Prefetch is

>     disabled if set to zero.

Reviewer 2: ```suggestion
    disabled if ``num-connector-io-threads`` is set to zero.
```
Suggest explicitly stating what to set to 0 to disable prefetch.

``async-data-cache-enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    * **Type** ``bool``
    * **Default value:** ``true``

>     Whether async data cache is enabled.  Disable this also disable split

Reviewer 2: ```suggestion
    Whether async data cache is enabled.  Setting ``async-data-cache-enabled`` to ``false`` disables split
```
Suggesting specifying what is done and its result.

    to ``false`` disables split prefetching in table scan.

> ``async-cache-ssd-gb``

Reviewer: consistency: async-data-cache-enabled vs. async-cache-ssd-gb - one says data-cache, the other just cache. Would be nice to fix.

Author: That will be a big change, rather not changing it here

Reviewer: Understood. Perhaps, file a GitHub issue about this.

Author: #21185

^^^^^^^^^^^^^^^^^^^^^^

    * **Type** ``integer``

>     * **Default value:** ``0``

Reviewer: Are we missing some other configs re: SSD. I assume one needs to specify rott directory or something, no?

Author: Just you will need to set SSD size in another property to indicate how much space is available

    Size of the SSD cache when async data cache is enabled.  Must be zero if

>     async data cache is disabled.

Reviewer 2: ```suggestion
    ``async-data-cache-enabled`` is disabled.
```
